samples/subsys/usb/audio/headphones_microphone/src/main.c

```diff
@@ -53,7 +53,7 @@ static void feature_update(const struct device *dev,
 	case USB_AUDIO_FU_MUTE_CONTROL:
 		break;
 	case USB_AUDIO_FU_VOLUME_CONTROL:
-		volume = *((int16_t *)(evt->val));
+		volume = UNALIGNED_GET((int16_t *)evt->val);
```

Comment:
Would it work with clang and why not just `memcpy(&volume, evt->val, sizeof(volume));`?

Reply:
My development environment is not convenient for testing clang (soc is a customized mips),  but zephyr also uses UNALIGNED_GET in many places, and `include/zephyr/toolchain/llvm.h` includes `include/zephyr/toolchain/gcc.h`, so clang should also be able to use it. If you have any doubts about this, I can resubmit using memcpy

Reply:
Looks like the __typeof__ should work there too https://clang.llvm.org/docs/UsersManual.html